app/src/main/java/com/conupods/MapsActivity.java

```diff
@@ -239,5 +275,26 @@ public static void setSearchViewOnClickListener(View v, View.OnClickListener lis
         }
     }
 
+    public synchronized void buildGoogleAPIClient() {
+        mGoogleAPIClient = new GoogleApiClient.Builder(this)
+                .addConnectionCallbacks(this)
+                .addOnConnectionFailedListener(this)
+                .addApi(LocationServices.API)
+                .build();
+        mGoogleAPIClient.connect();
+    }
```

Comment:
could this be abstracted

Reply:
What do you mean?

Reply:
like removed from mapActivity

Reply:
yes, but this i believe would be unnecessary refactoring since the api client belongs to the functionnality of this class